backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -0,0 +1,170 @@
+  private static AuthService authService;
+  private static RoleService roleService;
+  private int paginationSize;
+
+  public CollaboratorResourceMigration() {}
+
+  public void execute(int recordUpdateLimit) {
+    this.paginationSize = recordUpdateLimit;
+    App app = App.getInstance();
+    if (app.getAuthServerHost() != null && app.getAuthServerPort() != null) {
+      app.setAuthServerHost(app.getAuthServerHost());
+      app.setAuthServerPort(app.getAuthServerPort());
+
+      authService = new AuthServiceUtils();
+      roleService = new RoleServiceUtils(authService);
+    } else {
+      LOGGER.debug("AuthService Host & Port not found, OSS setup found");
+      return;
+    }
+
+    LOGGER.info("Migration start");
+    ModelDBUtils.registeredBackgroundUtilsCount();
+    try {
+      migrateProjects();
+      LOGGER.info("Projects done migration");
+      // migrateRepositories();
+      // LOGGER.info("Repositories done migration");
+    } catch (Exception ex) {
+      if (ex instanceof StatusRuntimeException) {
+        StatusRuntimeException exception = (StatusRuntimeException) ex;
+        if (exception.getStatus().getCode().value() == Code.PERMISSION_DENIED_VALUE) {
+          LOGGER.warn("CollaboratorResourceMigration Exception: {}", ex.getMessage());
+        } else {
+          LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+        }
+      } else {
+        LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+      }
+    } finally {
+      ModelDBUtils.unregisteredBackgroundUtilsCount();
+    }
+
+    LOGGER.info("Migration End");
+  }
+
+  private void migrateProjects() {
+    LOGGER.debug("Projects migration started");
+    Long count = getEntityCount(ProjectEntity.class);
+
+    int lowerBound = 0;
+    final int pagesize = this.paginationSize;
+    LOGGER.debug("Total Projects {}", count);
+
+    while (lowerBound < count) {
+
+      try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+
+        CriteriaQuery<ProjectEntity> criteriaQuery =
+            criteriaBuilder.createQuery(ProjectEntity.class);
+        Root<ProjectEntity> root = criteriaQuery.from(ProjectEntity.class);
+
+        CriteriaQuery<ProjectEntity> selectQuery =
+            criteriaQuery.select(root).orderBy(criteriaBuilder.asc(root.get("id")));
+
+        TypedQuery<ProjectEntity> typedQuery = session.createQuery(selectQuery);
+
+        typedQuery.setFirstResult(lowerBound);
+        typedQuery.setMaxResults(pagesize);
+        List<ProjectEntity> projectEntities = typedQuery.getResultList();
+
+        if (projectEntities.size() > 0) {
+          for (ProjectEntity project : projectEntities) {
+            try {
+              WorkspaceDTO workspaceDTO =
+                  roleService.getWorkspaceDTOByWorkspaceId(project.getWorkspace());
```

Comment:
Is `getWorkspace` populated? I'm pretty sure we never ended up populating it. Only the legacy ID is present today.

Reply:
we are always populating `workspace` field in the database https://github.com/VertaAI/modeldb/blob/fad1cd78aacc52c395179d649fb601534a142d7f/backend/src/main/java/ai/verta/modeldb/entities/ProjectEntity.java#L56. the legacy ID is not populated on the database

Reply:
Revert changes